Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -156,75 +181,77 @@ - (instancetype)initWithObjects:(id)firstObj {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (instancetype)initWithObjects:(id _Nonnull const[])objects count:(NSUInteger)cnt {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (instancetype)initWithObjects:(id _Nonnull const[])objects count:(NSUInteger)count {
+    if (self = [super init]) {
+        for (NSUInteger i = 0; i < count; ++i) {
+            [self _insertObject:objects[i]];
+        }
+    }
+    return self;
 }
-
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithOrderedSet:(NSOrderedSet*)orderedSet range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    auto values = orderedSet->setContainer;
+
+    NSMutableArray* arrayObjs = [NSMutableArray array];
+
+    for (id item : values) {
+        [arrayObjs addObject:item];
+    }
+
+    return [self initWithArray:arrayObjs range:range copyItems:flag];
 }
 
 /**
  @Status Stub
  @Notes
 */
 - (instancetype)initWithSet:(NSSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithSet:set copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithSet:(NSSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    NSMutableArray* arrayObjs = [NSMutableArray array];
+
+    for (id item : set) {
+        [arrayObjs addObject:item];
+    }
+    return [self initWithArray:arrayObjs range:NSMakeRange(0, [arrayObjs count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)init {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [super init];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (BOOL)containsObject:(id)object {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return std::find(setContainer.begin(), setContainer.end(), object) != setContainer.end();
```

Comment:
This is just going to compare pointers, isn't it? Not Objective-C object equality?